pkg/keystore/keystore.go

```diff
@@ -136,7 +145,52 @@ func (ks *Keystore) MaskTrustedKeySystemRoot(fingerprint string) (string, error)
 	return dst, ioutil.WriteFile(dst, []byte(""), 0644)
 }
 
-func (ks *Keystore) TrustedKeyPrefixExists(prefix string, r io.ReadSeeker) (bool, error) {
+// TrustKeyPrefixExists returns whether or not there exists 1 or more trusted
+// keys for a given prefix, or for any parent prefix.
+func (ks *Keystore) TrustedKeyPrefixExists(prefix string) (bool, error) {
+	acidentifier, err := types.NewACIdentifier(prefix)
+	if err != nil {
+		return false, err
+	}
+
+	pathNamesPrefix := []string{
+		// example: /etc/rkt/trustedkeys/prefix.d/coreos.com/etcd
+		path.Join(ks.LocalPrefixPath, acidentifier.String()),
+		// example: /usr/lib/rkt/trustedkeys/prefix.d/coreos.com/etcd
+		path.Join(ks.SystemPrefixPath, acidentifier.String()),
+	}
+
+	for _, p := range pathNamesPrefix {
+		_, err := os.Stat(p)
+		if os.IsNotExist(err) {
+			continue
+		}
+		if err != nil {
+			return false, fmt.Errorf("cannot check dir %q: %v", p, err)
+		}
+		files, err := ioutil.ReadDir(p)
+		if err != nil {
+			return false, fmt.Errorf("cannot list files in dir %q: %v", p, err)
+		}
+		for _, f := range files {
+			if f.Size() > 0 {
```

Comment:
`stat()` on directories have a non-zero reported size. This should probably check `f.IsDir()` before, to ignore directories.